### PR TITLE
Use nonroot user in notebook controller

### PR DESCRIPTION
**Pull request by MartinForReal**
Signed-off-by: MartinForReal <fanshangxiang@gmail.com>
Fixed: #5287

### CODE REVIEW

**Comment by MartinForReal** [comment]
related #5250

**Comment by MartinForReal** [comment]
ping ? @StefanoFioravanzo

**Comment by MartinForReal** [comment]
Could you please help review this pr? Thanks! @kimwnasptd

File: components/notebook-controller/Dockerfile [hunk 1]
```diff
@@ -6,11 +6,10 @@
 #
 # This is necessary because the Jupyter controller now depends on
 # components/common
-ARG GOLANG_VERSION=1.12.5
+ARG GOLANG_VERSION=1.13
```
Comment (naveensrinivasan) [inline components/notebook-controller/Dockerfile:9]: Why change the `go` version as part of this PR. IMO that should be a separate PR

**Comment by DavidSpek** [comment]
The version update for GO in this PR is also being done by https://github.com/kubeflow/kubeflow/pull/5394

**Comment by DavidSpek** [comment]
I just built the image without but without the update to the Go version (as this could cause conflicts with the PR mentioned above), and deployed it on my Kubeflow 1.2 installation and there don't seem to be any problems. Creating and deleting notebooks works as expected and no errors in the controller pod logs.

**Comment by yanniszark** [comment]
The Go version was updated as part of https://github.com/kubeflow/kubeflow/pull/5394.
@MartinForReal could you update your PR so that it only changes the image?

**Comment by DavidSpek** [comment]
For those interested, the difference between `base` and `static` is that `static` does not contain glibc, libssl, openssl. 
The `static` image does not have a `debug` flag, so no shell is present. If a shell would be needed the alternative is `base:debug-nonroot`. 
To check what PID the process runs as I built the image using `base:debug-nonroot` and this is the output of `ps aux`:
```
PID   USER     TIME  COMMAND
    1 nonroot   0:00 /manager
   48 nonroot   0:00 sh
   53 nonroot   0:00 ps aux
```

Other info from the BUILD file for the image that might be interesting to indicate what is different from the root image:
```
passwd_entry(       
name = "nonroot_user",       
gid = 65532,       
home = "/home/nonroot",       
info = "nonroot",       
shell = "/sbin/nologin",       
uid = 65532,       
username = "nonroot",     )
```

**Comment by DavidSpek** [comment]
@MartinForReal could you please remove the Go version update from this PR so that it can be merged?

**Comment by DavidSpek** [comment]
Closing in favor of https://github.com/kubeflow/kubeflow/pull/5428

/close
